code.js

```diff
@@ -1,3 +1,45 @@
-function quicksort(array) {
-    return array;
+// I was stuck on what the if/else check was supposed to be after the while loop, 
+// because I wasn't sure how it would repeat the iteration, so I got help from the
+// repo of the ClaytonBrown4741 because I noticed our code was similar!
+
+function swap(arr, first, second)
+{
+    let tmp = arr[first];
+    arr[first] = arr[second];
+    arr[second] = tmp;
+    return arr;
+}
+
+function quicksortIterative(arr, lo, hi)
+{
+    if(lo>=hi)
+    {
+        return arr;
+    }
+    let p;
+    let iteration = lo;
+    while(iteration < hi)
+    {
+        p = iteration;
+        if(arr[iteration] > arr[iteration + 1])
+        {
+            //From Jon Bentley's quicksort algorithm from the slides
+            for(i = lo + 1; i <= hi; i++)
+            {
+                if(arr[i] < arr[lo])
+                {swap(arr, ++p, i);}
+            }
+            swap(arr, lo, p);
+            iteration = 0;
+        }
+        else
+        { iteration++; }
```

Comment:
Doesn't this then work more like insertion sort than quicksort?

Reply:
I feel like yes and no, only because the partitioning code came from Jon Bentley's code on the slides and the while loop was used to keep iterating, but I can see how the if statement check is comparing the two elements to see if they are now sorted looks like insertion sort. That was the part I was stuck on and I had to consult other repos for help, but I can find a different way to partition and iterate if you want me to?

Reply:
If you could find a way to process the two partitions separately that would be great!